instrumentation/jdbc/library/src/test/java/io/opentelemetry/instrumentation/jdbc/OpenTelemetryConnectionTest.java

```diff
@@ -0,0 +1,235 @@
+import java.sql.SQLException;
+import java.sql.Statement;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+class OpenTelemetryConnectionTest {
+
+  @RegisterExtension
+  static final InstrumentationExtension testing = LibraryInstrumentationExtension.create();
+
+  @Test
+  void testVerifyCreateStatement() throws SQLException {
+    Instrumenter<DbRequest, Void> instrumenter =
+        createStatementInstrumenter(testing.getOpenTelemetry());
+    DbInfo dbInfo = getDbInfo();
+    OpenTelemetryConnection connection =
+        new OpenTelemetryConnection(new TestConnection(), dbInfo, instrumenter);
+    String query = "SELECT * FROM users";
+    Statement statement = connection.createStatement();
+
+    testing.runWithSpan(
+        "parent",
+        () -> {
+          assertThat(statement.execute(query)).isTrue();
+        });
+
+    testing.waitAndAssertTraces(
+        trace ->
+            trace.hasSpansSatisfyingExactly(
+                span -> span.hasName("parent").hasKind(SpanKind.INTERNAL).hasNoParent(),
+                span ->
+                    span.hasName("SELECT my_name.users")
+                        .hasKind(SpanKind.CLIENT)
+                        .hasParent(trace.getSpan(0))
+                        .hasAttributesSatisfyingExactly(
+                            equalTo(DbIncubatingAttributes.DB_SYSTEM, dbInfo.getSystem()),
+                            equalTo(DbIncubatingAttributes.DB_NAME, dbInfo.getName()),
+                            equalTo(DbIncubatingAttributes.DB_USER, dbInfo.getUser()),
+                            equalTo(DbIncubatingAttributes.DB_STATEMENT, query),
+                            equalTo(DbIncubatingAttributes.DB_OPERATION, "SELECT"),
+                            equalTo(DbIncubatingAttributes.DB_SQL_TABLE, "users"),
+                            equalTo(ServerAttributes.SERVER_ADDRESS, dbInfo.getHost()),
+                            equalTo(ServerAttributes.SERVER_PORT, dbInfo.getPort()))));
+
+    statement.close();
+    connection.close();
+  }
+
+  @Test
+  void testVerifyCreateStatementReturnsOtelWrapper() throws Exception {
+    OpenTelemetry ot = OpenTelemetry.propagating(ContextPropagators.noop());
+    Instrumenter<DbRequest, Void> instrumenter = createStatementInstrumenter(ot);
+    OpenTelemetryConnection connection =
+        new OpenTelemetryConnection(new TestConnection(), DbInfo.DEFAULT, instrumenter);
+
+    assertThat(connection.createStatement()).isInstanceOf(OpenTelemetryStatement.class);
+    assertThat(connection.createStatement(0, 0)).isInstanceOf(OpenTelemetryStatement.class);
+    assertThat(connection.createStatement(0, 0, 0)).isInstanceOf(OpenTelemetryStatement.class);
+
+    Field instrumenterField = OpenTelemetryStatement.class.getDeclaredField("instrumenter");
+    instrumenterField.setAccessible(true);
+
+    assertThat(instrumenterField.get(connection.createStatement())).isEqualTo(instrumenter);
```

Comment:
I'd move the test to `internal` package, the same as the tested `OpenTelemetryConnection` and `OpenTelemetryStatement` classes. Since the `instrumenter` field is `protected` you won't need to use reflection to read it when the test is in the same package.

Reply:
Did you try `assertThat(((OpenTelemetryStatement) connection.createStatement()).instrumenter).isEqualTo(instrumenter);`

Reply:
ahhh ok, yea that works, thank you!